tests/ServerRequestTest.php

```diff
@@ -519,4 +519,19 @@ public function testNullAttribute()
         $this->assertSame([], $requestWithoutAttribute->getAttributes());
         $this->assertSame('different-default', $requestWithoutAttribute->getAttribute('name', 'different-default'));
     }
+    
+    public function testExtend()
+    {
+        $actual = FooExtend::fromGlobals();
+        $this->assertInstanceOf('GuzzleHttp\\Tests\\Psr7\\FooExtend', $actual);
```

Comment:
This can be written in an cleaner way, as `assertInstanceOf(FooExtend::class, $actual)`.

This avoids needing to manually resolve namespace paths or escape characters.

Reply:
done

Reply:
@Krinkle it is possible, because support php5.4 https://travis-ci.org/guzzle/psr7/builds/349217118

Reply:
Hmm... I see. I wasn’t expecting PHP 5.4 support here in the master branch. The oldest is expect at this point is PHP 5.6, or, of supporting current oldstable Debian, then PHP 5.5 as well.

Thanks for considering! Maybe another time.